functions/security/idtoken.go

```diff
@@ -0,0 +1,52 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package security contains samples for securely calling functions.
+package security
+
+// [START functions_bearer_token]
+import (
+	"context"
+	"fmt"
+	"io"
+
+	"google.golang.org/api/idtoken"
+)
+
+// callFunction makes a request to the provided functionURL with an
+// authenticated client.
+func callFunction(w io.Writer, functionURL string) error {
```

Comment:
If this does not need to conform to the GCF function signature, would be great to update and use github.com/GoogleCloudPlatform/golang-samples/tree/master/run/authenticating for both Cloud Run and Cloud Functions.

Reply:
I would be fine updating that sample and using it for both.

Reply:
To stay inline with other languages, I think I would like to keep this a separate sample if that is okay. I think it will provide a clearer picture to consumers of the docs.

Reply:
We have other cases where we have the same sample used in multiple places in the docs, would like to understand the differences other than "typical practice" that degrade the clarity of information.